### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by maltfield** [oasisprotocol/oasis-core-rosetta-gateway#169]
Hello,

Our server keeps crashing immediately after launching it with the following error

```
oasis_1  | ts=2021-03-01T16:16:45.246179191Z level=info module=oasis-rosetta-gateway caller=main.go:157 msg="waiting for node socket to appear..." socket_path=/data/internal.sock
oasis_1  | ts=2021-03-01T16:16:46.246362407Z level=info module=oasis-rosetta-gateway caller=main.go:157 msg="waiting for node socket to appear..." socket_path=/data/internal.sock
oasis_1  | ts=2021-03-01T16:16:47.247397561Z level=debug module=oasis caller=oasis.go:134 msg="Establishing connection" grpc_addr=unix:/data/internal.sock
oasis_1  | panic: runtime error: invalid memory address or nil pointer dereference
oasis_1  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x1337bb2]
oasis_1  | 
oasis_1  | goroutine 158 [running]:
oasis_1  | github.com/tendermint/tendermint/node.(*Node).Switch(...)
oasis_1  | 	github.com/tendermint/tendermint@v0.33.6/node/node.go:1116
oasis_1  | github.com/oasisprotocol/oasis-core/go/consensus/tendermint/full.(*fullService).GetStatus(0xc000375520, 0x1f51280, 0xc01c141dd0, 0x19387a0, 0x1, 0x0)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/consensus/tendermint/full/full.go:809 +0x1d2
oasis_1  | github.com/oasisprotocol/oasis-core/go/control.(*nodeController).GetStatus(0xc00ffded20, 0x1f51280, 0xc01c141dd0, 0x1f63600, 0xc00ffded20, 0x0)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/control/control.go:85 +0x63
oasis_1  | github.com/oasisprotocol/oasis-core/go/control/api.handlerGetStatus.func1(0x1f51280, 0xc01c141dd0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/control/api/grpc.go:230 +0x6d
oasis_1  | github.com/oasisprotocol/oasis-core/go/common/grpc/auth.UnaryServerInterceptor.func1(0x1f51280, 0xc01c141dd0, 0x0, 0x0, 0xc00fe16b40, 0xc00fe16b60, 0x2ba0ca0, 0x1b0b1a0, 0x2ba0340, 0x1f51280)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/common/grpc/auth/auth.go:43 +0x20f
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x1f51280, 0xc01c141dd0, 0x0, 0x0, 0x2ba0340, 0x1f51280, 0xc01c141dd0, 0x1f6c900)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x63
oasis_1  | github.com/oasisprotocol/oasis-core/go/common/grpc.serverUnaryErrorMapper(0x1f51280, 0xc01c141dd0, 0x0, 0x0, 0xc00fe16b40, 0xc00fe16b80, 0xc00fe16c20, 0xc01c141da0, 0x0, 0x0)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/common/grpc/errors.go:94 +0x55
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x1f51280, 0xc01c141dd0, 0x0, 0x0, 0x1bcbdb8, 0xd, 0xc0002c6d50, 0x24)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x63
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1(0x1f51280, 0xc01c141d10, 0x0, 0x0, 0xc00fe16b40, 0xc00fe16ba0, 0x4c02c6, 0x603d136f, 0xee526f5, 0x23fe4dd20d06a7)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/tracing/opentracing/server_interceptors.go:34 +0x10b
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x1f51280, 0xc01c141d10, 0x0, 0x0, 0xc0100942c8, 0x5, 0x203003, 0x0)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x63
oasis_1  | github.com/oasisprotocol/oasis-core/go/common/grpc.(*grpcLogAdapter).unaryLogger(0xc000fb3bf0, 0x1f51280, 0xc01c141d10, 0x0, 0x0, 0xc00fe16b40, 0xc00fe16bc0, 0x0, 0x0, 0x203007, ...)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/common/grpc/grpc.go:190 +0x16b
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x1f51280, 0xc01c141d10, 0x0, 0x0, 0x30, 0x30, 0xc00d8abb28, 0x419b38)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:25 +0x63
oasis_1  | github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x1f51280, 0xc01c141d10, 0x0, 0x0, 0xc00fe16b40, 0xc00fe16b60, 0xc01c141d10, 0xc00d8abba0, 0x4cb406, 0x1a9b480)
oasis_1  | 	github.com/grpc-ecosystem/go-grpc-middleware@v1.2.2/chain.go:34 +0xd7
oasis_1  | github.com/oasisprotocol/oasis-core/go/control/api.handlerGetStatus(0x1a82280, 0xc00ffded20, 0x1f51280, 0xc01c141d10, 0xc01c3a26c0, 0xc000fb3c80, 0x1f51280, 0xc01c141d10, 0x0, 0x0)
oasis_1  | 	github.com/oasisprotocol/oasis-core/go/control/api/grpc.go:232 +0x123
oasis_1  | google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ee1c0, 0x1f650a0, 0xc00d8a4300, 0xc001140200, 0xc00ffdedb0, 0x2a082a8, 0x0, 0x0, 0x0)
oasis_1  | 	google.golang.org/grpc@v1.32.0/server.go:1194 +0x522
oasis_1  | google.golang.org/grpc.(*Server).handleStream(0xc0002ee1c0, 0x1f650a0, 0xc00d8a4300, 0xc001140200, 0x0)
oasis_1  | 	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd05
oasis_1  | google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc01c13fcd0, 0xc0002ee1c0, 0x1f650a0, 0xc00d8a4300, 0xc001140200)
oasis_1  | 	google.golang.org/grpc@v1.32.0/server.go:859 +0xa5
oasis_1  | created by google.golang.org/grpc.(*Server).serveStreams.func1
oasis_1  | 	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd
oasis_1  | level=error module=grpc/client caller=grpc.go:242 ts=2021-03-01T16:16:47.289911088Z msg="request failed" method=/oasis-core.NodeController/GetStatus req_seq=1 rsp="unsupported value type" err="rpc error: code = Unavailable desc = transport is closing"
oasis_1  | ts=2021-03-01T16:16:47.289968336Z level=debug module=oasis caller=oasis.go:147 msg="Failed to get status from node" err="rpc error: code = Unavailable desc = transport is closing"
oasis_1  | ts=2021-03-01T16:16:47.290805284Z level=error module=oasis-rosetta-gateway caller=main.go:175 msg="failed to obtain chain ID from Oasis node" err="failed to get status from node: rpc error: code = Unavailable desc = transport is closing"
ubuntu_oasis_1 exited with code 1
```

Also, it appears from the blockchain explorer's past ~900 pages of transactions that that there is some issue causing most transactions to fail.

 * https://www.oasisscan.com/transactions

The issue appears to have started either late `2021-02-27` or early `201-02-28` UTC.

Please advise how to start our node and how to debug what is the input causing the `panic` ... `invalid memory address or nil pointer dereference` ... `segmentation violation`

**Comment by kostko** [comment]
This seems like it is due to a bug in Oasis Core when an early `GetStatus` query happens that was fixed a while back. Which version of Oasis Core are you running?

**Comment by maltfield** [comment]
```bash-5.0# oasis-node --version
Software version: 20.12.2
Consensus:
  Consensus protocol version: 2.0.0
  Tendermint Core version:    0.34.0
  ABCI library version:       0.17.0
Runtime:
  Host protocol version:      1.0.0
  Committee protocol version: 1.0.0
Go toolchain version: 1.15.5
bash-5.0# 
```

**Comment by kostko** [comment]
Please try upgrading to Oasis Core 20.12.4.

**Comment by maltfield** [comment]
I don't see anything related to `GetStatus` or `invalid memory address or nil pointer dereference` issues marked as being fixed in `20.12.4`'s changelog:

 * https://github.com/oasisprotocol/oasis-core/blob/v20.12.4/CHANGELOG.md 

But I took your advice, upgraded to `20.12.4`, and now the bootlooping docker image is stable. Thank you.

**Comment by ptrus** [comment]
Was fixed in `20.12.3` https://github.com/oasisprotocol/oasis-core/blob/v20.12.4/CHANGELOG.md#bug-fixes-1 (go/consensus/tendermint: Fix early GetStatus() with initial height > 1 ).

Great that the issue was resolved!
